source/setup.py

```diff
@@ -110,80 +92,60 @@ def getRecursiveDataFiles(dest,source,excludes=()):
 
-setup(
-	name = name,
-	version=version,
-	description=description,
-	url=url,
-	classifiers=[
-'Development Status :: 3 - Alpha',
-'Environment :: Win32 (MS Windows)',
-'Topic :: Adaptive Technologies'
-'Intended Audience :: Developers',
-'Intended Audience :: End Users/Desktop',
-'License :: OSI Approved :: GNU General Public License (GPL)',
-'Natural Language :: English',
-'Programming Language :: Python',
-'Operating System :: Microsoft :: Windows',
-],
-	cmdclass={"py2exe": py2exe},
-	windows=[
-		{
-			"script":"nvda.pyw",
-			"dest_base":"nvda_noUIAccess",
-			"uiAccess": False,
-			"icon_resources":[(1,"images/nvda.ico")],
-			"other_resources": [], # Populated at run time
-			"version":formatBuildVersionString(),
-			"description":"NVDA application",
-			"product_name":name,
-			"product_version":version,
-			"copyright": NVDAcopyright,
-			"company_name":publisher,
-		},
-		# The nvda_uiAccess target will be added at runtime if required.
-		{
-			"script": "nvda_slave.pyw",
-			"uiAccess": False,
-			"icon_resources": [(1,"images/nvda.ico")],
-			"other_resources": [], # Populated at run time
-			"version":formatBuildVersionString(),
-			"description": name,
-			"product_name":name,
-			"product_version": version,
-			"copyright": NVDAcopyright,
-			"company_name": publisher,
-		},
-		{
-			"script": "nvda_eoaProxy.pyw",
-			# uiAccess will be enabled at runtime if appropriate.
-			"uiAccess": False,
-			"icon_resources": [(1,"images/nvda.ico")],
-			"other_resources": [], # Populated at run time
-			"version":formatBuildVersionString(),
-			"description": "NVDA Ease of Access proxy",
-			"product_name":name,
-			"product_version": version,
-			"copyright": NVDAcopyright,
-			"company_name": publisher,
-		},
-	],
+
+def _genManifestTemplate(shouldHaveUIAccess: bool) -> Tuple[int, int, bytes]:
+	return (
+		RT_MANIFEST,
+		1,
+		(_manifestTemplate % {"uiAccess": shouldHaveUIAccess}).encode("utf-8")
+	)
+
+
+_py2ExeWindows = [
+	{
+		"script": "nvda.pyw",
+		"dest_base": "nvda_noUIAccess",
+		"icon_resources": [(1, "images/nvda.ico")],
+		"other_resources": [_genManifestTemplate(shouldHaveUIAccess=False)],
+	},
+	# The nvda_uiAccess target will be added at runtime if required.
+	{
+		"script": "nvda_slave.pyw",
+		"icon_resources": [(1, "images/nvda.ico")],
+		"other_resources": [_genManifestTemplate(shouldHaveUIAccess=False)],
+	},
+]
+if _partialArgs.uiAccess:
+	_py2ExeWindows.insert(1, {
+		"script": "nvda.pyw",
+		"dest_base": "nvda_uiAccess",
+		"icon_resources": [(1, "images/nvda.ico")],
+		"other_resources": [_genManifestTemplate(shouldHaveUIAccess=True)],
+	})
+
+
+freeze(
+	version_info={
+		"version": formatBuildVersionString(),
+		"description": description,
+		"product_name": name,
+		"product_version": version,
+		"copyright": NVDAcopyright,
+		"company_name": publisher,
+	},
+	windows=_py2ExeWindows,
 	console=[
 		{
 			"script": os.path.join("..", "include", "nvda_dmp", "nvda_dmp.py"),
 			"uiAccess": False,

```

Comment:
There is no need to specify `uiAccess` here any longer. This line can be removed.

Reply:
can you explain why? We still need to specify UI Access in the manifest for each exe

Reply:
We certainly do, but we need to do this in the same way as for all other exe's i.e. by adding `_genManifestTemplate(shouldHaveUIAccess=False)` to the `other_resources` for the nvda-dmp binary.

Reply:
My mistake, this was a misunderstanding over your review comment